src/v/redpanda/admin_server.cc

```diff
@@ -2017,4 +2029,71 @@ void admin_server::register_transaction_routes() {
 
           co_return ss::json::json_return_type(ans);
       });
+
+    ss::httpd::transaction_json::delete_partition.set(
+      _server._routes,
+      [this](std::unique_ptr<ss::httpd::request> req)
+        -> ss::future<ss::json::json_return_type> {
+          auto transaction_id = req->param["transactional_id"];
+
+          auto namespace_from_req = req->get_query_param("namespace");
+          auto topic_from_req = req->get_query_param("topic");
+
+          auto partition_str = req->get_query_param("partition_id");
+          int64_t partition;
+          try {
+              partition = std::stoi(partition_str);
+          } catch (...) {
+              throw ss::httpd::bad_param_exception(fmt_with_ctx(
```

Comment:
I don't think you want to use `fmt_with_ctx` when returning errors to users (same for other places below)